ssg/build_cpe.py

```diff
@@ -198,17 +206,35 @@ class CPEALPlatform(object):
     prefix = "cpe-lang"
     ns = PREFIX_TO_NS[prefix]
 
-    def __init__(self, id):
-        self.id = id
-        self.test = None
+    def __init__(self, expression, algebra, cpe_products):
+        self.test = algebra.parse(expression, simplify=True)
+        self.id = "cpe_platform_" + self.test.as_id()
+        self.cpe_products = cpe_products
+        self._replace_cpe_names(self.test)
+
+    def _replace_cpe_names(self, exp):
+        if isinstance(exp, CPEALFactRef):
```

Comment:
It would be better to call something like `exp.replace_cpe_name()` that would either perform the operation, or it would recurse. The current approach has the `CPEALPlatform` class that knows too much - instead of telling `exp` to do the right thing, we either treat it in a special way, or we recurse on its children.